spring-ai-core/src/main/java/org/springframework/ai/chat/client/ChatClient.java

```diff
@@ -74,9 +74,9 @@ static Builder builder(ChatModel chatModel, ObservationRegistry observationRegis
 
 	ChatClientRequestSpec prompt();
 
-	ChatClientPromptRequestSpec prompt(String content);
+	ChatClientRequestSpec prompt(String content);
```

Comment:
I didn't think that allowing the 'sealed' prompts to participate in the rest of the fluent api would be this straightforward, nice!

I just added docs today wrt to the overloads being 'sealed'  so I'll remove that and seems like we can delete the interface `ChatClientPromptRequestSpec`